src/java/org/streamingpool/core/service/impl/PoolContent.java

```diff
@@ -84,4 +88,15 @@ public PoolContent() {
     private void addStreamHooks() {
         activeStreams.put(NEW_STREAM_HOOK, newStreamHook);
     }
+
+    /**
+     * Add a dependency to the tree from source to parent
+     */
+    public void addDependency(StreamId<?> source, StreamId<?> parent) {
+        dependencies.addDependency(source, parent);
+    }
+
+    public StreamDependencyTree dependencyTree() {
```

Comment:
If I am right, the Stream dependency tree is a mutable object, right? I wonder if it would be good to return here an immutable copy?

Reply:
let's discuss in the other comment if we move this field to the instrumentation service directly. Then the immutable copy could be an option. At the moment the mutability is just controlled via interface, which is not perfect we know :)